src/basic/process-util.c

```diff
@@ -2099,7 +2100,28 @@ int posix_spawn_wrapper(
 #if HAVE_PIDFD_SPAWN
         _cleanup_close_ int pidfd = -EBADF;
 
+        /* If we have observed that SPAWN_CGROUP is not supported, remove the flag */
+        if (!setcgroup_supported && FLAGS_SET(flags, POSIX_SPAWN_SETCGROUP)) {
```

Comment:
Please just skip the opening of cgroup altogether if we know in advance it would be ignored anyway.

Reply:
Yeah, please see the above code in my comment.

Reply:
yes, the opening of the cgorup is what is slow, and if it doesn't work we really shouldn't bother.